src/library/basesqltablemodel.h

```diff
@@ -139,6 +139,23 @@ class BaseSqlTableModel : public QAbstractTableModel, public TrackModel {
         }
         int m_column;
         Qt::SortOrder m_order;
+        
+        friend QTextStream& operator >> (QTextStream &in, SortColumn &sc) {
```

Comment:
I am afraid we hit a conceptual issue. :-/ 
The current solution is a simple serialize without ans sanity checks. The same we have with the header state.  
Since this is not that critical data, that "could" work as it is, including all issues we already heave.  
If not, we should should probably ditch the SettingsDAO entirely and move to mixxx.cfg, which allows us to parse and check the values before using. .. and is human readable.  
@jmigual and @rryan: What do you think about it?